### PR TITLE
Add support for `/` and run CleanUserAgentInfoEntry method on _clientVersion

### PR DESCRIPTION
**Pull request by ruant**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by dsgouda** [comment]
@shahabhijeet PTAL

File: src/SdkCommon/ClientRuntime/ClientRuntime/ServiceClient.cs [hunk 1]
```diff
@@ -502,7 +504,7 @@ public bool SetUserAgent(string productName, string version)
         /// <returns></returns>
         private string CleanUserAgentInfoEntry(string infoEntry)
         {
-            Regex pattern = new Regex("[~`!@#$%^&*(),<>?{} ]");            
+            Regex pattern = new Regex("[~`!@#$%^&*(),<>?{} \/]");
```
Comment (shahabhijeet) [inline src/SdkCommon/ClientRuntime/ClientRuntime/ServiceClient.cs]: @ruant can you provide a scenario or an example where you are encountering an error?
Also add a test that covers the "/" scenario that you are encountering.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SdkCommon/ClientRuntime/ClientRuntime/ServiceClient.cs (hunk 1 above)
Comment (ruant) [reply src/SdkCommon/ClientRuntime/ClientRuntime/ServiceClient.cs]:
```
System.FormatException: The format of value '6.1.12+Branch.release/6.1.12.Sha.3d0b9f3f9bb053111ba468326221f21b95124a3c' is invalid.
   at System.Net.Http.Headers.HeaderUtilities.CheckValidToken(String value, String parameterName)
   at System.Net.Http.Headers.ProductHeaderValue..ctor(String name, String version)
   at Microsoft.Rest.ServiceClient`1.InitializeHttpClient(HttpClientHandler httpClientHandler, DelegatingHandler[] handlers)
```

I'll whip up a test that covers it.

**Comment by shahabhijeet** [comment]
@ruant please re-open this PR after you have addressed the feedback on this PR.

**Pull request closed without merging**